cbmc/proofs/IotMqtt_UnsubscribeAsync/Makefile

```diff
@@ -0,0 +1,117 @@
+DEF += -DAWS_IOT_MQTT_SERVER_MAX_TOPIC_LENGTH=$(AWS_IOT_MQTT_SERVER_MAX_TOPIC_LENGTH)
+
+# this should be smaller than topic_length_max for coverage
+# _IotMqtt_SubscriptionPacketSize coverage at line 702
+MQTT_MAX_REMAINING_LENGTH=6
+DEF += -DMQTT_MAX_REMAINING_LENGTH=$(MQTT_MAX_REMAINING_LENGTH)
+
+################################################################
+# Function omitted from the proof
+
+ABSTRACTIONS += --remove-function-body IotLog_Generic
+
+# Functions removed for coverage
+ABSTRACTIONS += --remove-function-body _IotMqtt_AddSubscriptions
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeSubscribe
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeSubscribeCommon
+ABSTRACTIONS += --remove-function-body _matchEndWildcards
+ABSTRACTIONS += --remove-function-body _matchWildcards
+ABSTRACTIONS += --remove-function-body _mqttOperation_match
+ABSTRACTIONS += --remove-function-body _mqttSubscription_setUnsubscribe
+ABSTRACTIONS += --remove-function-body _packetMatch
+ABSTRACTIONS += --remove-function-body _topicFilterMatch
+ABSTRACTIONS += --remove-function-body _validateQos
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceClose
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceReceive
+#ABSTRACTIONS += --remove-function-body _IotMqtt_PublishSetDup
+#ABSTRACTIONS += --remove-function-body _IotMqtt_RemoveSubscriptionByPacket
+#ABSTRACTIONS += --remove-function-body _checkRetryLimit
+#ABSTRACTIONS += --remove-function-body _scheduleNextRetry
+
+################################################################
+
+OBJS += $(ENTRY)_harness.goto
+OBJS += $(MQTT)/cbmc/proofs/mqtt_state.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_api.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_helper.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_operation.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_serialize.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_subscription.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_validate.goto
+
+DEF += -DCBMC_STUB_REMOVEALLMATCHES=1
+DEF += -DCBMC_UNSUBSCRIBEASYNC_PROOF=1
+
+# allow mqtt mallocs to fail for coverage
+DEF += -include mqtt_state.h
+DEF += -DIotMqtt_MallocOperation=malloc_can_fail
+DEF += -DIotMqtt_MallocMessage=malloc_can_fail
+
+################################################################
+# Loop unwinding
+
+LOOP += harness.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_RemoveSubscriptionByTopicFilter.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribe.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribeCommon.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _encodeRemainingLength.0:2
+LOOP += _IotMqtt_SubscriptionPacketSize.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _validateSubscription.0:$(TOPIC_LENGTH_MAX)
+LOOP += allocate_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += valid_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
```

Comment:
I usually group them in lexicographical order for readability.

Reply:
Done.